packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3292,6 +3294,86 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
+  /// Handles [ScrollIntent] by scrolling the [Scrollable] inside of
+  /// [EditableText].
+  void _scroll(ScrollIntent intent) {
+    if (intent.type != ScrollIncrementType.page) {
+      return;
+    }
+
+    final ScrollPosition position = _scrollController.position;
+    if (widget.maxLines == 1) {
+      _scrollController.jumpTo(position.maxScrollExtent);
+      return;
+    }
+
+    final ScrollableState? state = _scrollableKey.currentState as ScrollableState?;
+    final double increment = ScrollAction.getIncrement(state!, intent);
+    final double destination = (position.pixels + increment).clamp(
+      position.minScrollExtent,
+      position.maxScrollExtent,
+    );
+    if (destination == position.pixels) {
+      return;
+    }
+    _scrollController.jumpTo(destination);
+  }
+
+  /// Extend the selection down by page if the `forward` parameter is true, or
+  /// up by page otherwise.
+  void _extendSelectionByPage(ExtendSelectionByPageIntent intent) {
```

Comment:
This PR now handles vertical runs.  Let me know if I've missed anything though.

Reply:
Actually I was mistaken and it doesn't, but maybe it doesn't need to because it only moves the cursor during selection.  Let me know what you think.

Reply:
To explain this here for anyone that reads this thread: We decided to not handle shift + pageup/down in this PR, so it only does scrolling and does not need vertical runs.  That is handled in another PR by @tgucio that has since been merged.